pandas/core/arrays/datetimes.py

```diff
@@ -308,18 +308,33 @@ def _from_sequence_not_strict(
     ):
```

Comment:
Would be great to add a docstring for `_from_sequence_not_strict` while here. Not completely obvious what "not strict" means relative to `_from_sequence`

Reply:
good idea.  will add that in the follow-up branch